test/common/extensions/cert_validator/platform_bridge/platform_bridge_cert_validator_test.cc

```diff
@@ -0,0 +1,314 @@
+    EXPECT_NE(main_thread_id_, std::this_thread::get_id());
+
+    // Make sure the cert was converted correctly.
+    const Buffer::InstancePtr buffer = Data::Utility::toInternalData(*certs);
+    const auto digest = Common::Crypto::UtilitySingleton::get().getSha256Digest(*buffer);
+    EXPECT_EQ(TEST_SAN_DNS2_CERT_256_HASH, Hex::encode(digest));
+    return mock_validator_->validate(certs, size, hostname);
+  }
+
+  static void cleanup() {
+    // Validate must be called on the worked thread, not the main thread.
+    EXPECT_NE(main_thread_id_, std::this_thread::get_id());
+    mock_validator_->cleanup();
+  }
+
+  static std::unique_ptr<MockValidator> mock_validator_;
+  static std::thread::id main_thread_id_;
+
+  Api::ApiPtr api_;
+  Event::DispatcherPtr dispatcher_;
+  Stats::TestUtil::TestStore test_store_;
+  SslStats stats_;
+  Ssl::MockCertificateValidationContextConfig config_;
+  std::string empty_string_;
+  SSLContextPtr ssl_ctx_;
+  TestSslExtendedSocketInfo ssl_extended_info_;
+  CertValidator::ExtraValidationContext validation_context_;
+  Network::TransportSocketOptionsConstSharedPtr transport_socket_options_;
+  std::unique_ptr<MockValidateResultCallback> callback_;
+  bool is_server_;
+  envoy_cert_validator platform_validator_;
+};
+
+std::unique_ptr<MockValidator> PlatformBridgeCertValidatorTest::mock_validator_;
+std::thread::id PlatformBridgeCertValidatorTest::main_thread_id_;
+
+INSTANTIATE_TEST_SUITE_P(TrustMode, PlatformBridgeCertValidatorTest,
+                         testing::ValuesIn({CertificateValidationContext::VERIFY_TRUST_CHAIN,
+                                            CertificateValidationContext::ACCEPT_UNTRUSTED}));
+
+TEST_P(PlatformBridgeCertValidatorTest, NoConfig) {
+  EXPECT_ENVOY_BUG(
+      { PlatformBridgeCertValidator validator(nullptr, stats_, &platform_validator_); },
+      "Invalid certificate validation context config.");
+}
+
+TEST_P(PlatformBridgeCertValidatorTest, NonEmptyCaCert) {
+  std::string ca_cert = "xyz";
+  EXPECT_CALL(config_, caCert()).WillRepeatedly(ReturnRef(ca_cert));
+  EXPECT_CALL(config_, certificateRevocationList()).WillRepeatedly(ReturnRef(empty_string_));
+  EXPECT_CALL(config_, trustChainVerification()).WillRepeatedly(Return(GetParam()));
+
+  EXPECT_ENVOY_BUG(
+      { PlatformBridgeCertValidator validator(&config_, stats_, &platform_validator_); },
+      "Invalid certificate validation context config.");
+}
+
+TEST_P(PlatformBridgeCertValidatorTest, NonEmptyRevocationList) {
+  std::string revocation_list = "xyz";
+  EXPECT_CALL(config_, caCert()).WillRepeatedly(ReturnRef(empty_string_));
+  EXPECT_CALL(config_, certificateRevocationList()).WillRepeatedly(ReturnRef(revocation_list));
+  EXPECT_CALL(config_, trustChainVerification()).WillRepeatedly(Return(GetParam()));
+
+  EXPECT_ENVOY_BUG(
+      { PlatformBridgeCertValidator validator(&config_, stats_, &platform_validator_); },
+      "Invalid certificate validation context config.");
```

Comment:
are there details in the log which would let us differentiate this invalid from the above invalid?